### PR TITLE
Fix issues found in compose metrics

### PR DESCRIPTION
**Pull request by patrickhoette**
Fixed the issues found in the compose metrics. Mainly caused by passing lists around.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -59,4 +59,5 @@ subprojects {
 
 tasks.register("clean", Delete::class) {
     delete(rootProject.buildDir)
+    delete(file("$rootProject/compose-metrics"))
```
Comment (mcabrita0) [inline build.gradle.kts:62]: Could add it to .gitignore instead? Might be useful to keep the results between cleans

**Review by mcabrita0 — APPROVED** [review]

File: markymark/src/main/java/com/m2mobi/markymark/composer/DefaultMarkyMarkComposer.kt [hunk 2]
```diff
@@ -366,7 +369,7 @@ open class DefaultMarkyMarkComposer : MarkyMarkComposer {
         ListItem(
             modifier = modifier.fillParentMaxWidth(),
             type = node.type,
-            children = node.children,
+            children = node.children.toImmutableList(),
```
Comment (mcabrita0) [inline markymark/src/main/java/com/m2mobi/markymark/composer/DefaultMarkyMarkComposer.kt:372]: This is inside a `@Composable` and will run every recomposition, needs a remember

File: build.gradle.kts (hunk 1 above)
Comment (patrickhoette) [reply build.gradle.kts:62]: I added it to the .gitignore. This is just to clean everything up properly when running the clean task. Project only takes ~16s to compile so I don't think keeping them between cleans is realy relevant

File: markymark/src/main/java/com/m2mobi/markymark/composer/DefaultMarkyMarkComposer.kt (hunk 2 above)
Comment (patrickhoette) [reply markymark/src/main/java/com/m2mobi/markymark/composer/DefaultMarkyMarkComposer.kt:372]: Good catch. `node.children` already is an `ImmutableList` seeing as the Android Lint PR switches to the Kotlin Immutable Collections, I will fix it there

**Pull request merged**